hax-lib/Cargo.toml

```diff
@@ -17,3 +17,6 @@ num-traits = { version = "0.2.15", default-features = false }
 [features]
 default = ["macros"]
 macros = ["dep:hax-lib-macros"]
+
+[lints.rust]
+unexpected_cfgs = { level = "warn", check-cfg = ['cfg(hax)'] }
```

Comment:
There's a new lint that shouts if you use an undeclared feature or `cfg` flag. This silences it on the many `cfg(hax)` checks. An alternative would be to use `cfg(feature = "hax")` instead and decalre that feature like the ones above.